app/src/main/java/com/calendar/Activity/CountDays.java

```diff
@@ -0,0 +1,153 @@
+        numberOfFreeDaysText = findViewById(R.id.numberOfFreeDays);
+        numberOfWorkingDaysText = findViewById(R.id.numberOfWorkingDays);
+    }
+
+    private void setMaterialDatePicker() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.DAY_OF_MONTH, 1);
+        long monthStart = calendar.getTimeInMillis();
+        calendar.set(Calendar.DAY_OF_MONTH, calendar.getActualMaximum(Calendar.DAY_OF_MONTH));
+        long monthEnd = calendar.getTimeInMillis();
+
+        materialDatePicker = MaterialDatePicker.Builder
+                .dateRangePicker()
+                .setSelection(new Pair<>(monthStart, monthEnd))
+                .build();
+        setMaterialDatePickerListener();
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private void setMaterialDatePickerListener() {
+        materialDatePicker.addOnPositiveButtonClickListener(selection -> {
+            dateRangeText.setText(materialDatePicker.getHeaderText());
+            Date startDate = new Date(selection.first);
+            Date endDate = new Date(selection.second);
+            LocalDate startLocalDate = dateToLocalDate(startDate);
+            LocalDate endLocalDate = dateToLocalDate(endDate);
+            countDays(startLocalDate, endLocalDate);
+        });
+    }
+
+    public void selectDateRange(View view) {
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private LocalDate dateToLocalDate(Date date) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        int year = calendar.get(Calendar.YEAR);
+        int month = calendar.get(Calendar.MONTH) + 1;
+        int day = calendar.get(Calendar.DAY_OF_MONTH);
+        return LocalDate.of(year, month, day);
+    }
+
+    public void countDays(LocalDate startDate, LocalDate endDate) {
+        numberOfFreeDays = 0;
+        long numberOfDays = ChronoUnit.DAYS.between(startDate, endDate) + 1;
+
+        databaseReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                for (LocalDate date = startDate; date.isBefore(endDate.plusDays(1)); date = date.plusDays(1)) {
+                    if (isFree(dataSnapshot, date)) {
+                        numberOfFreeDays++;
+                    }
+                }
+                numberOfFreeDaysText.setText(valueOf(numberOfFreeDays));
+                numberOfWorkingDaysText.setText(valueOf(numberOfDays - numberOfFreeDays));
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+                Log.e(TAG, "calendarClicked - onCancelled", error.toException());
+            }
+        });
+    }
+
+    private boolean isFree(DataSnapshot dataSnapshot, LocalDate date){
+        if (isWeekend(date.getDayOfWeek()) || isHoliday(dataSnapshot,valueOf(date))) {
```

Comment:
It can be simplified. If statement is true return true, if is false return false... Mbe just return the statement's value.